ferrocene/ci/scripts/persist-between-jobs.sh

```diff
@@ -50,7 +58,20 @@ case "$1" in
         # Call `zstd` separately to be able to use all cores available (`-T0`)
         # and the lowest compression level possible, to speed the compression
         # as much as possible.
-        tar c --exclude build/metrics.json $@ | zstd -1 -T0 | aws s3 cp - "$(s3_url "${CIRCLE_JOB}")"
+        #
+        # On Windows we have to pass `-f -`, otherwise tar will write to \\.\tape0
+        # rather than stdout by default.
+        if [[ "${OSTYPE}" = "msys" ]]; then
+            "${TAR}" -cf- --exclude build/metrics.json \
+                --exclude build/x86_64-pc-windows-msvc/stage0-sysroot/lib/rustlib/rustc-src/rust --exclude build/x86_64-pc-windows-msvc/stage0-sysroot/lib/rustlib/src/rust --exclude build/x86_64-pc-windows-msvc/stage1/lib/rustlib/rustc-src/rust --exclude build/x86_64-pc-windows-msvc/stage1/lib/rustlib/src/rust --exclude build/x86_64-pc-windows-msvc/stage2/lib/rustlib/rustc-src/rust --exclude build/x86_64-pc-windows-msvc/stage2/lib/rustlib/src/rust --exclude build/host \
```

Comment:
This is.... unfortunate, but the symlinks we generate in the tarball end up having a lot of issues. We need to manually exclude them.

We should consider swapping to `.7z` or some other more Windows friendly format.

The good news is we don't really use this since we didn't split build/dist.